packages/design-system/src/components/VerticalNav/VerticalNavItem.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from 'react';
+import VerticalNav from './VerticalNav';
+import VerticalNavItemLabel from './VerticalNavItemLabel';
+import classNames from 'classnames';
+import uniqueId from 'lodash/uniqueId';
+
+export type VerticalNavItemComponent = React.ReactElement<any> | any | ((...args: any[]) => any);
+
+export interface VerticalNavItemProps {
+  /**
+   * @hide-prop This gets passed through from the parent VerticalNav to a nested VerticalNav
+   */
+  _selectedId?: string;
+  /**
+   * Aria label for the toggle button when the sub-navigation is collapsed
+   */
+  ariaCollapsedStateButtonLabel?: string;
+  /**
+   * Aria label for the toggle button when the sub-navigation is expanded
+   */
+  ariaExpandedStateButtonLabel?: string;
+  /**
+   * Additional classes to be added to the root element
+   */
+  className?: string;
+  /**
+   * When provided, this will render the passed in component. This is useful when
+   * integrating with React Router's `<Link>` or using your own custom component.
+   */
+  component?: VerticalNavItemComponent;
+  /**
+   * Whether or not the item's sub-navigation is in a collapsed state by default
+   */
+  defaultCollapsed?: boolean;
+  /**
+   * Called when the link is clicked, with the following arguments:
+   * [`SyntheticEvent`](https://facebook.github.io/react/docs/events.html),
+   * `id`, `url`.
+   * This takes precedence over the `VerticalNav` `onLinkClick` prop
+   */
+  onClick?: (...args: any[]) => any;
+  /**
+   * Called when this item's subnav is collapsed or expanded, with the
+   * following arguments: `id`, `collapsed`
+   */
+  onSubnavToggle?: (...args: any[]) => any;
+  /**
+   * Optional identifier. This can be handy if you're passing in an
+   * `onClick` handler. A unique ID will be generated if one isn't provided.
+   */
+  id?: string;
+  /**
+   * An array of nested `VerticalNavItem` data objects to be rendered in a
+   * sub-navigation list.
+   */
+  items?: any[];
+  /**
+   * Text to render for this nav item
+   */
+  label: React.ReactNode;
+  /**
+   * A URL to navigate to if this item is a link
+   */
+  url?: string;
+  /**
+   * If this item is currently selected
+   */
+  selected?: boolean;
+}
+
+export const VerticalNavItem = (props: VerticalNavItemProps): React.ReactElement => {
+  const id = props.id || uniqueId('VerticalNavItem_');
+  const subnavId = `${id}__subnav`;
+
+  const [collapsed, setCollapsed] = useState(props.defaultCollapsed);
+
+  // Refactor from `componentDidUpdate` to `useEffect` using this example:
+  // https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
+  React.useEffect(() => {
+    if (props.onSubnavToggle) {
+      props.onSubnavToggle(props.id, collapsed);
+    }
+  }, [collapsed]);
+
```

Comment:
Since `collapsed` is only updated by one line in this component (line 96), consider moving the body of the use effect to directly after the `setCollapsed` call in `handleToggleClick`.

`useEffect` might be necessary if the component doesn't know when `collapsed` might be changing, but here it is extraneous because the component knows when `setCollapsed` is being called